docs/source/use-cases.rst

```diff
@@ -0,0 +1,375 @@
+
+This solution is particularly attractive because it stays within the Python
+ecosystem, and uses the speed and algorithm set of Pandas_, a tool with which
+the analyst is already very comfortable.
+
+.. code-block:: python
+
+   from dask.distributed import Executor
+   e = Executor('cluster-address:8786')
+
+   import dask.dataframe as dd
+   df = dd.read_csv('hdfs://data/2016-*.*.csv', parse_dates=['timestamp'])
+   df.groupby(df.timestamp.dt.hour).value.mean().compute()
+
+.. _HDFS3: http://hdfs3.readthedocs.io/en/latest/
+
+
+Directories of custom tabular files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The same analyst has a bunch of files of a custom format not supported by
+Dask.dataframe, or perhaps these files are in a directory structure that
+encodes important information about his data (such as the date or other
+metadata.)  They use :doc:`dask.delayed<delayed>` to teach Dask.dataframe how
+to load the data and then pass into dask.dataframe for tabular algorithms.
+
+* Example Notebook: https://gist.github.com/mrocklin/e7b7b3a65f2835cda813096332ec73ca
+
+
+JSON data
+~~~~~~~~~
+
+Data Engineers with click stream data from a website or mechanical engineers
+with telemetry data from mechanical instruments have large volumes of data in
+JSON or some other semi-structured format.  They use :doc:`dask.bag<bag>` to
+manipulate many Python objects in parallel either on their personal machine,
+where they stream the data through memory or across a cluster.
+
+.. code-block:: python
+
+   import dask.bag as db
+   import json
+
+   records = db.read_text('data/2015-*-*.json').map(json.loads)
+   records.filter(lambda d: d['name'] == 'Alice').pluck('id').frequencies()
+
+
+Ad Hoc Examples
+---------------
+
+The large collections (array, dataframe, bag) are wonderful when they fit the
+application, for example if you want to perform a groupby on a directory of CSV
+data.  However several parallel computing applications don't fit neatly into
+one of these higher level abstractions.  Fortunately, Dask provides a wide
+variety of ways to parallelize more custom applications.  These use the same
+machinery as the arrays and dataframes, but allow the user to develop custom
+algorithms specific to their problem.
+
+Embarrassingly parallel computation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A programmer has a function that they want to run many times on different
+inputs.  Their function and inputs might use arrays or dataframes internally,
+but conceptually their problem isn't a single large array or dataframe.
+
+They want to run these functions in parallel on their laptop while they prototype
+but they also intend to eventually use an in-house cluster.  They wrap their
+function in :doc:`dask.delayed<delayed>` and let the appropriate dask scheduler
+parallelize and load balance the work.
+
+.. code-block:: python
+
+   def process(data):
+      ...
+      return ...
+
+   # results = [process(x) for x in inputs]  # sequential
```

Comment:
originally commented out because we've left sequential processing.  I've now added lots of little sections for the different options, including one for sequential processing.